### PR TITLE
issue: 749133 Fix setsockopt IGMP struct size check

### PR DESCRIPTION
**Pull request by DanielLibenson**
Signed-off-by: Daniel Libenson <daneilli@mellanox.com>

### CODE REVIEW

File: src/vma/sock/sockinfo_udp.cpp [hunk 1]
```diff
@@ -926,7 +926,7 @@ int sockinfo_udp::setsockopt(int __level, int __optname, __const void *__optval,
 						break;
 					}
 
-					if ((NULL == __optval) || (__optlen > sizeof(struct ip_mreq_source) || (__optlen < sizeof(struct ip_mreq)))) {
+					if ((NULL == __optval) || (__optlen < sizeof(struct ip_mreq))) {
```
Comment (rosenbaumalex) [inline src/vma/sock/sockinfo_udp.cpp]: @DanielLibenson later on (Line #934) you cast to `struct ip_mreq_source` so you must verify __optlen is big enough before using that part on memory.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/466/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/451/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/452/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/468/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/453/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/469/ for details (Mellanox internal link).

**Pull request merged**